docs/streams/upgrade-guide.html

```diff
@@ -59,6 +59,16 @@ <h1>Upgrade Guide and API Changes</h1>
         For Kafka Streams 0.10.0, broker version 0.10.0 or higher is required.
     </p>
 
+    <p>
+        Another important thing to keep in mind: in deprecated <code>KStreamBuilder</code> class, when a <code>KTable</code> is created from a source topic via <code>KStreamBuilder.table()</code>, its materialized state store
+        will reuse the source topic as its changelog topic for restoring, and will disable logging to avoid appending new updates to the source topic; in the <code>StreamsBuilder</code> class introduced in 1.0, this behavior has been changed
+        accidentally: we still reuse the source topic as the changelog topic for restoring, but will also create a separate changelog topic to append the update records from source topic to. In the 2.0 release, we have fixed this issue and now users
+        and choose whether or not to reuse the source topic based on the <code>StreamsConfig#TOPOLOGY_OPTIMIZATION</code>: if you want to continue reusing the source topic as the changelog topic for restoration when upgrading to 2.0.0, from either
```

Comment:
`and now users and choose` -- not sure what you wanted to write :)

Reply:
Also, can users really "choose". It might be better to give strict instructions. When upgrading from KStreamBuilder to 2.0 StreamsBuilder you must enable optimization. If you upgrade from `StreamsBuilder` to 2.0 you need to disable optimization (ie, do nothing special, as this is the default -- we should mention this?)

Reply:
ack